test/e2e/common/setup-kubestellar.sh

```diff
@@ -109,25 +131,25 @@ echo "wds1 created."
 : Run OCM transport controller in a pod
 :
 if [ "$use_release" != true ]; then
-  cd "${SRC_DIR}/../../.." ## go up to KubeStellar directory
+  pushd "${SRC_DIR}/../../.." ## go up to KubeStellar directory
   KUBESTELLAR_DIR="$(pwd)"
   OCM_TRANSPORT_PLUGIN_RELEASE="0.1.7"
   curl -sL https://github.com/kubestellar/ocm-transport-plugin/archive/refs/tags/v${OCM_TRANSPORT_PLUGIN_RELEASE}.tar.gz | tar xz
   cd ocm-transport-plugin-${OCM_TRANSPORT_PLUGIN_RELEASE}
   OCM_TRANSPORT_PLUGIN_DIR="$(pwd)"
   pwd
   echo "replace github.com/kubestellar/kubestellar => ${KUBESTELLAR_DIR}/" >> go.mod
-  go mod tidy # TODO to be deleted next time we bump ocm transport release (done in ocm transport makefile)
   IMAGE_TAG=${OCM_TRANSPORT_PLUGIN_RELEASE} make ko-build-local
   kind load --name kubeflex docker-image ko.local/transport-controller:${OCM_TRANSPORT_PLUGIN_RELEASE} # load local image to kubeflex
   cd "${KUBESTELLAR_DIR}"
   pwd
   rm -rf ${OCM_TRANSPORT_PLUGIN_DIR}
   echo "running ocm transport plugin..."
-  kubectl config use-context kind-kubeflex ## transport deployment script assumes it runs within kubeflex context
+  kubectl config use-context "$HOSTING_CONTEXT" ## transport deployment script works with current context
```

Comment:
Wouldn't it be better to remove this and change the deployment script to get a context name parameter, and have that script use the context directly?  
Setting context has always bad side effects , for example an error may leave us pointing to a different context than the one we started with, etc..

Reply:
Done.